engine/fresh/manifest.ts

```diff
@@ -175,37 +166,50 @@ export const $live = <T extends DecoManifest>(m: T): T => {
           },
         )
         : {}; // if block has no adapt so it's not considered a resolver.
+      const recover = adapted[localRef(blk.type, danglingModuleTS)] ??
+        adapted[localRef(blk.type, danglingModuleTSX)] ??
+        blk.defaultDanglingRecover;
       return [
         { ...currMan, [blk.type]: decorated },
         { ...currMap, ...adapted, ...previews },
-        blk.defaultDanglingRecover
+        (recover as Resolver | undefined)
           ? [...recovers, {
-            recoverable: (type: string) => {
-              const splitted = type.split("/");
-              // check if there's any segment on the same name of the block
-              return splitted.some((segment) => segment === blk.type); //FIXME (mcandeia) this is not a straightforward solution
-            },
-            recover: adapted[localRef(blk.type, danglingModuleTS)] ??
-              adapted[localRef(blk.type, danglingModuleTSX)] ??
-              blk.defaultDanglingRecover,
+            recoverable: resolverIsBlock(blk),
+            recover,
           } as DanglingRecover]
           : recovers,
       ];
     },
     [m, {}, []] as [DecoManifest, ResolverMap<FreshContext>, DanglingRecover[]],
   );
   context.site = siteName();
-  const provider = getProvider();
+  const provider = getComposedConfigStore(
+    context.namespace!,
+    context.site,
+    context.siteId,
+  );
+  context.configStore = provider;
   const resolver = new ConfigResolver<FreshContext>({
     resolvers: { ...resolvers, ...defaultResolvers, preview },
-    getResolvables: provider.get.bind(provider),
+    getResolvables: provider.state.bind(provider),
     danglingRecover: recovers.length > 0
       ? buildDanglingRecover(recovers)
       : undefined,
   });
+
+  if (shouldCheckIntegrity) {
+    resolver.getResolvables().then(
+      (resolvables: Record<string, Resolvable>) => {
+        integrityCheck(resolver.getResolvers(), resolvables);
+      },
+    );
+  }
   // should be set first
   context.configResolver = resolver;
   context.manifest = newManifest;
+  console.log(
+    `Starting live: siteId=${context.siteId} site=${context.site}`,
+  );
```

Comment:
If we are to ditch siteId (as number), could we be more "depreacted-ly" here?

Reply:
yes, we can! I think it should be addressed in the next steps when blocks UI was released